addons/website_sale/models/res_partner.py

```diff
@@ -18,8 +18,31 @@ def _compute_last_website_so_id(self):
             if website and not is_public:
                 partner.last_website_so_id = SaleOrder.search([
                     ('partner_id', '=', partner.id),
+                    ('pricelist_id', '=', partner.property_product_pricelist.id),
                     ('website_id', '=', website.id),
                     ('state', '=', 'draft'),
                 ], order='write_date desc', limit=1)
             else:
                 partner.last_website_so_id = SaleOrder  # Not in a website context or public User
+
+    @api.onchange('property_product_pricelist')
+    def _onchange_property_product_pricelist(self):
+        self.ensure_one()
+
+        open_order_count = self.env['sale.order'].search_count([
```

Comment:
use search(, limit=1)

Reply:
After that change `open_order_count` is quite misleading variable name, I changed it